pymc3/step_methods/metropolis.py

```diff
@@ -120,9 +120,10 @@ def astep(self, q0):
                 q = (q0 + delta).astype(int)
             else:
                 delta[self.discrete] = round(delta[self.discrete], 0).astype(int)
-                q = q0 + delta
+                q = (q0 + delta)
         else:
-            q = q0 + delta
+            q0 = q0.astype(theano.config.floatX)
+            q = (q0 + delta).astype(theano.config.floatX)
 
         q_new = metrop_select(self.delta_logp(q, q0), q, q0)
```

Comment:
Actually this is just a python function (contrary to what I said earlier, I misremembered because we made `delta_logp` a theano function, and that function should only be compiled once).